docs/proposals/metrics.md

```diff
@@ -0,0 +1,78 @@
+# Proposal: Metrics proposal
+
+Last updated: 17.11.2020
+
+## Abstract
+
+Expose metrics of the baremetal CSI plugin that will give an idea of how different parts of the system performs.
+
+## Background
+As part of scalability testing, we should understand what parts are much slower and easily find the bottlenecks. After instrumenting the baremetal CSI plugin with metrics we'll be able to see hard it is working and what it's actually doing.
+
+## Proposal
+
+Expose metrics in Prometheus format with HTTP server at "/metrics" endpoint on every component:
+ - Node
+ - Controller  
+ - Scheduler Extender    
+ - Drive manager 
+
+The HTTP server will be optional since it can cause some security issues(i.e. vulnerabilities with HTTP server)
+Default port: 8787
+
+Metrics:
+
+Metric name                           | Metric type |                          Labels/tags                                     |              Description
```

Comment:
@painhardcore is it possible to collect metrics from sidecar components - csi-provisioner, csi-attacher, etc?

Reply:
@mishoyama yes, as long they support metrics endpoint with Prometheus format.
From the first look - they have it.
[provisioned as example](https://github.com/kubernetes-csi/external-provisioner)
```
--metrics-address: The TCP network address where the prometheus metrics endpoint will run (example: :8080 which corresponds to port 8080 on local host). The default is empty string, which means metrics endpoint is disabled.

--metrics-path: The HTTP path where prometheus metrics will be exposed. Default is /metrics.
```